### ISSUE TITLE
False positive for InvalidArgument Error with \DateTimeImmutable

### ISSUE THREAD

**Issue by jrushlow** [vimeo/psalm#2948]
False `Error` by psalm when passing a `\DateTimeImmutable::class` to a method that expects a `\DateTimeInterface::class`. This appeared after composer update:

`vimeo/psalm  dev-master a2a5071    dev-master a0da735 A static analysis tool for finding errors in PHP applications`

```
// ClassA
public function test(\DateTimeInterface $time): void
{
}
```

```
// ClassB
$a = new ClassA();
$a->test(new \DateTimeImmutable());
```

The above results in:
```
ERROR: InvalidArgument - src/A.php:x:x - Argument 1 of ClassA::test expects DateTimeInterface, DateTimeImmutable provided
        $a->(new \DateTimeImmutable());
```

May be related to issue #2933 although I have not had a chance to dig into it.

**Comment by jrushlow** [comment]
https://psalm.dev/r/da2890ba9e

**Comment by muglug** [comment]
ah right, thanks, broken an hour ago by 40a7a11

**Comment by jrushlow** [comment]
thanks for the quick fix!
